python/ql/lib/semmle/python/security/dataflow/LdapInjection.qll

```diff
@@ -0,0 +1,87 @@
+/**
+ * Provides a taint-tracking configuration for detecting LDAP injection vulnerabilities
+ */
+
+import python
+import semmle.python.Concepts
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A taint-tracking configuration for detecting LDAP injections.
+ */
+class LDAPInjectionFlowConfig extends TaintTracking::Configuration {
+  LDAPInjectionFlowConfig() { this = "LDAPInjectionFlowConfig" }
+
+  override predicate isSource(DataFlow::Node source, DataFlow::FlowState state) {
+    source instanceof RemoteFlowSource and
+    state instanceof Unsafe
+  }
+
+  override predicate isSink(DataFlow::Node sink, DataFlow::FlowState state) {
+    sink = any(LdapExecution ldap).getBaseDn() and
+    (state instanceof Unsafe or state instanceof UnsafeForDn)
+    or
+    sink = any(LdapExecution ldap).getFilter() and
+    (state instanceof Unsafe or state instanceof UnsafeForFilter)
+  }
+
+  override predicate isBarrier(DataFlow::Node node, DataFlow::FlowState state) {
+    // All additional flow steps signify a state change.
+    //    (n, `state`) --> (`node`, s)
+    // Thus, if a node in `state` transitions to `node` and a new state,
+    // then `state` should be blocked at `node`.
+    isAdditionalFlowStep(_, state, node, _)
+  }
+
+  override predicate isAdditionalFlowStep(
+    DataFlow::Node node1, DataFlow::FlowState state1, DataFlow::Node node2,
+    DataFlow::FlowState state2
```

Comment:
is there a reason you're not using the typical `nodeFrom` -> `nodeTo` naming?

Reply:
That comes from the data flow API. I would change it (but I will do a two-config version instead first).